app/controllers/addFinancialInstitution/registeredBusiness/IsTheAddressCorrectController.scala

```diff
@@ -48,28 +53,53 @@ class IsTheAddressCorrectController @Inject() (
 
   val form = formProvider()
 
-  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen retrieveCtUTR() andThen getData andThen requireData).async {
     implicit request =>
-      val preparedForm = request.userAnswers.get(IsTheAddressCorrectPage) match {
-        case None        => form
-        case Some(value) => form.fill(value)
+      request.ctutr match {
+        case Some(utr) =>
+          regService.fetchAddress(utr).flatMap {
+            address =>
+              for {
+                updatedAnswers <- Future.fromTry(request.userAnswers.set(FetchedRegisteredAddressPage, address))
+                result <- sessionRepository.set(updatedAnswers).map {
+                  _ =>
+                    val preparedForm = request.userAnswers.get(IsTheAddressCorrectPage) match {
+                      case None        => form
+                      case Some(value) => form.fill(value)
+                    }
+                    Ok(view(preparedForm, mode, getFinancialInstitutionName(request.userAnswers), formatAddressBlock(address).value))
```

Comment:
Added a comment on the `formatAddressBlock` function but in short - can we pass the address to the view as an Address and let it decide how to render / format it?